weDevsOfficial/dokan
# subscription product category fix

### All Submissions:






### Changes proposed in this Pull Request:



Closes #1662
Closes https://github.com/weDevsOfficial/dokan-pro/issues/1963

Dokan Pro PR: https://github.com/weDevsOfficial/dokan-pro/pull/1969

### How to test the changes in this Pull Request:

1. Create a Dokan subscription product with some category restriction on it, subscribe to a vendor on it, and try to add a product in a category you will see only the restricted category will show.
2. When you restrict categories you will only top-level categories.
3. Check responsive design in the multi-step category in the vendor dashboard.



### Changelog entry

- **Fix:** Categories when restrict categories from vendor subscription module.
- **Fix:** Responsive mobile version in category UI.

### PR Self Review Checklist:

* Code is not following code style guidelines
* Bad naming: make sure you would understand your code if you read it a few months from now.
* KISS: Keep it simple, Sweetie (not stupid!).
* DRY: Don't Repeat Yourself.
* Code that is not readable: too many nested 'if's are a bad sign.
* Performance issues
* Complicated constructions that need refactoring or comments: code should almost always be self-explanatory.
* Grammar errors.


### FOR PR REVIEWER ONLY:

> As a reviewer, your feedback should be focused on the idea, not the person. Seek to understand, be respectful, and focus on constructive dialog.

> As a contributor, your responsibility is to learn from suggestions and iterate your pull request should it be needed based on feedback. Seek to collaborate and produce the best possible contribution to the greater whole.
